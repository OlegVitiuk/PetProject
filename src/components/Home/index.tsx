import { Button, Checkbox, Form, Icon, Input } from "antd";
import { FormComponentProps } from "antd/lib/form";
import { IAppState } from "store";
import * as React from "react";
import { connect } from "react-redux";
import styles from "./signUp.module.scss";

interface IProps {}

interface IState {}

class Home extends React.Component<IProps, IState> {
  public render() {
    return <div>HOME PAGE</div>;
  }
}

// Grab the characters from the store and make them available on props
const mapStateToProps = (store: IAppState) => {
  return {};
};

export default connect(mapStateToProps)(Home);
