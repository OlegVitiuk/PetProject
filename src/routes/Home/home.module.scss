.homeWrapper {
  display: flex;
  margin: 50px auto 20px;
  padding: 5px 20px;
  width: 100%;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.goalsContainer {
  display: flex;
  flex-wrap: wrap;
  margin-top: 30px;
}

.goalsHeader {
  width: 400px;
}

.homeHeader {
  margin-bottom: 10px;
  text-align: center;
}

.goalName {
  font-family: "Inconsolata";
  max-width: 200px;
  text-overflow: ellipsis;
  overflow: hidden;
}

.styledListItem {
  display: flex;
  justify-content: space-between;
}

.deleteIcon {
  color: #ff4136;
}

.goalCard {
  margin: 10px;
  width: 300px;
  display: flex;
  flex-direction: column;
  border: 1px solid Maroon;
  border-radius: 7px;
  min-height: 80px;
  padding: 5px;
  justify-content: space-between;
}
.goalHeader {
  display: flex;
  justify-content: space-between;
  align-items: center;

  span {
    margin-right: auto;
  }
}

.more {
  color: maroon;
  font-size: 20px;
}

.noData {
  margin-top: 10px;
  font-family: "Inconsolata";
  font-size: 18px;
  color: maroon;
  font-weight: bold;
}
