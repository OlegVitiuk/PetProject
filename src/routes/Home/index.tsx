import { addGoal, removeGoal } from "actions/goalActions";
import { Icon, Input, Spin, Tag } from "antd";
import { Action } from "redux";
import * as React from "react";
import { connect } from "react-redux";
import { firestoreConnect } from "react-redux-firebase";
import { bindActionCreators, compose } from "redux";
import { ThunkDispatch } from "redux-thunk";
import { AppState } from "store";
import { IGoal } from "types/goal";
import styles from "./home.module.scss";
import { IGoalState } from "types/goal";

const { Search } = Input;

interface HomePageProps {
  goalsData: IGoalState;
  authorId: string;
}

interface HomePageState {}

interface LinkStateProps {
  goalsData: IGoalState;
  authorId: string;
}

interface LinkDispatchProps {
  addGoal: (goal: IGoal) => void;
  removeGoal: (id: string) => void;
}

type Props = HomePageProps & LinkDispatchProps;
type State = HomePageState & LinkStateProps;

class Home extends React.Component<Props, State> {
  public render() {
    const { goalsData } = this.props;

    const goalsIds = goalsData && Object.keys(goalsData);

    return (
      <div className={styles.homeWrapper}>
        <div className={styles.goalsHeader}>
          <h1 className={styles.homeHeader}>Add a goal</h1>
          <Search
            placeholder="Enter your goal"
            enterButton="Add"
            allowClear={true}
            size="large"
            onSearch={this.processGoal}
          />
        </div>
        <div className={styles.goalsContainer}>
          {goalsData &&
            goalsIds.map(goalId => {
              const goalData = goalsData[goalId];
              if (goalData) {
                return (
                  <div className={styles.goalCard} key={goalId}>
                    <div className={styles.goalHeader}>
                      <span className={styles.goalName}>{goalData.name}</span>
                      <Tag
                        key={goalData.authorId}
                        color={goalData.paid ? "green" : "red"}
                      >
                        {goalData.paid ? "paid" : "not paid"}
                      </Tag>
                      <Icon
                        type="delete"
                        onClick={() => this.deleteGoal(goalId)}
                        className={styles.deleteIcon}
                      />
                    </div>
                    <Icon type="down-circle" className={styles.more} />
                  </div>
                );
              }
              return null;
            })}
          {!goalsData && <span className={styles.noData}>No data</span>}
        </div>
      </div>
    );
  }

  private processGoal = (value: string): void => {
    const { addGoal, authorId } = this.props;

    addGoal({
      name: value,
      paid: false,
      price: 100,
      authorId
    });
  };

  private deleteGoal = (id: string): void => {
    const { removeGoal } = this.props;

    removeGoal(id);
  };
}

export default compose<any>(
  connect(
    (state: AppState): LinkStateProps => ({
      goalsData: state.firestore.data.Goals,
      authorId: state.firebase.auth.uid
    }),
    (dispatch: ThunkDispatch<any, any, Action>): LinkDispatchProps => ({
      addGoal: bindActionCreators(addGoal, dispatch),
      removeGoal: bindActionCreators(removeGoal, dispatch)
    })
  ),
  firestoreConnect((props: any) => {
    return [
      { collection: "Goals", where: [["authorId", "==", props.authorId]] }
    ];
  })
)(Home);
