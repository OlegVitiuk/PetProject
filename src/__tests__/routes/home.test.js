import React from 'react';
import { shallow } from 'enzyme';
import Home from 'routes/Home';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureMockStore([thunk]);

describe('Home route', () => {
  let store = {};
  let wrapper;

  beforeEach(() => {
    store = mockStore({
      firebase: {
        auth: {
          uid: 'asdfasfd'
        }
      },
      firestore: {
        data: {
          Goals: {}
        }
      }
    });
  });

  it('render Home component', () => {
    const wrapper = shallow(<Home store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
