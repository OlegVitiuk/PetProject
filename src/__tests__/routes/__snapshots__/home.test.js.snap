// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home route render Home component 1`] = `ShallowWrapper {}`;
